contracts/seed/Seed.sol

```diff
@@ -127,112 +131,124 @@ contract Seed {
       * @param _tokens                Array containing two params:
                                         - The address of the seed token being distributed.
       *                                 - The address of the funding token being exchanged for seed token.
-      * @param _softAndHardCap         Array containing two params:
+      * @param _softHardReq           Array containing two params:
                                         - the minimum funding token collection threshold in wei denomination.
                                         - the highest possible funding token amount to be raised in wei denomination.
+                                        - var to store the amount of seed to be distributed after calculation here
       * @param _price                 The price in wei of fundingTokens when exchanged for seedTokens.
       * @param _startTime             Distribution start time in unix timecode.
       * @param _endTime               Distribution end time in unix timecode.
-      * @param _vestingDuration       Vesting period duration in days.
-      * @param _vestingCliff          Cliff duration in days.
-      * @param _permissionedSeed         Set to true if only whitelisted adresses are allowed to participate.
+      * @param _vestingDuration       Vesting period duration in seconds.
+      * @param _vestingCliff          Cliff duration in seconds.
+      * @param _permissionedSeed      Set to true if only whitelisted adresses are allowed to participate.
       * @param _fee                   Success fee expressed as a % (e.g. 2 = 2% fee)
     */
     function initialize(
         address _beneficiary,
         address _admin,
         address[] memory _tokens,
-        uint256[] memory _softAndHardCap,
+        uint256[] memory _softHardReq,
         uint256 _price,
```

Comment:
let's remove price from initializer and just calculate it from `hardCap / seedAmountAtStart`

Reply:
okay

Reply:
seedAmountAtStart includes our fee, this will give us an incorrect price, to calculate the price, we first need to deduct the fee and then, find it.

Reply:
or even why are we trying to do this, can't we just calculate the amount seedToken sent at the beginning that have been sent, because the parameter isn't changing.

Reply:
Added to Issue #160 , This PR is getting very big. Closing this request.